lib/redis_memo/memoize_method.rb

```diff
@@ -81,14 +81,63 @@ def self.extract_dependencies(ref, *method_args, &depends_on)
 
   def self.get_or_extract_dependencies(ref, *method_args, &depends_on)
     if RedisMemo::Cache.local_dependency_cache
-      RedisMemo::Cache.local_dependency_cache[ref] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on][method_args] ||= extract_dependencies(ref, *method_args, &depends_on)
+      RedisMemo::Cache.local_dependency_cache[ref.class] ||= {}
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on] ||= {}
+      named_args = exclude_anonymous_args(depends_on, ref, method_args)
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on][named_args] ||= extract_dependencies(ref, *method_args, &depends_on)
     else
       extract_dependencies(ref, *method_args, &depends_on)
     end
   end
 
+  # We only look at named method parameters in the dependency block in order to define its dependent
+  # memos and ignore anonymous parameters, following the convention that nil or :_ is an anonymous parameter.
+  # Example:
+  # ```
+  #    def method(param1, param2)
+  #    end
+  #
+  #    memoize_method :method do |_, _, param2|`
+  #      depends_on RedisMemo::Memoizable.new(param2: param2)
+  #    end
+  # ```
+  #  `exclude_anonymous_args(depends_on, ref, [1, 2])` returns [2]
+  def self.exclude_anonymous_args(depends_on, ref, args)
+    return [] if depends_on.parameters.empty? or args.empty?
+
+    positional_args = []
+    kwargs = {}
+    named_rest = false
+    depends_on_args = [ref] + args
+    options = depends_on_args.extract_options!&.clone
```

Comment:
neat!
```suggestion
    options = depends_on_args.clone.extract_options!
```
`extract_options` cannot be nil right?

Reply:
Instead I just created a deep copy of the original args array